core/src/main/java/jenkins/model/item_category/ItemCategoryConfigurator.java

```diff
@@ -103,12 +103,41 @@ public static String getDescription(@Nonnull TopLevelItemDescriptor descriptor)
         throw new IllegalStateException("At least, a default value must exist for description field");
     }
 
+    /**
+     * Provides the effective clazz for the requested item or null if this configurator doesn't have one.
+     *
+     * @param descriptor the item to categorize
+     *
+     * @return A string or null
+     */
+    @CheckForNull
+    protected abstract String getEffectiveClazzFor(@Nonnull TopLevelItemDescriptor descriptor);
+
+    /**
+     * Finds the weight specified by the first configurator.
+     * If none can be found a empty string with {@code descriptor.clazz.getName();} is returned. {@see DefaultConfigurator#getEffectiveClazzFor}.
```

Comment:
Got it, the `DefaultConfigurator` takes care of that. Still not an empty string.